### DISCUSSION TITLE
running-async-tasks-on-app-startup-in-asp-net-core-part-2/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#284] [Announcements]
# running-async-tasks-on-app-startup-in-asp-net-core-part-2/

In this post I show two approaches for running tasks asynchronously on startup. Each approach allow registering tasks using DI, but have different trade offs

https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-part-2/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 23 2019, 09:15, in reply to [Paulo Morgado](https://disqus.com/by/paulo_morgado/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/284#discussioncomment-6054296)</em>

---

<p>Very much depends on the task, but mostly it just makes things a bit easier. Once all the tasks are run, all the tasks are complete. Otherwise you have to handle task completion more carefully. I'm going to touch on this in the next post.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 25 2019, 09:45, in reply to [Ulrik B](https://disqus.com/by/ulrikb/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/284#discussioncomment-6054301)</em>

---

<p>I wouldn't recommend that. a) if you're doing async tasks, then you're doing sync-over-async which, while probably not a problem here, is a bad practice b) You shouldn't be running tasks in the constructor of a random interface. Especially not async tasks.</p><p>If you're running small, sync tasks then there are other options if you're not a fan of the approach I described here 🙂 <a href="https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-part-1/" rel="nofollow noopener" title="https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-part-1/">https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-part-1/</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 26 2019, 07:29, in reply to [Dave Ikin](https://disqus.com/by/daveikin/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/284#discussioncomment-6056214)</em>

---

<p>Yeah I'm afraid you're out of luck in that case. Probably easier to use the other approach, e.g. Thomas Levesque's NuGet package instead 🙂</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at October 18 2019, 06:13, in reply to [Adam Stapleton](https://disqus.com/by/adamstapleton/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/284#discussioncomment-6056216)</em>

---

<p>Thanks! Yep, it's a lot easier in 3.0: <a href="https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-3/" rel="nofollow noopener" title="https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-3/">https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-3/</a></p>

**Comment by satrapu** [comment]
Thanks, Andrew, for this very helpful post - even though it has been published almost 5 years ago, it's still relevant and really helped me in refactoring executing application startup logic from sync to async!
